### DISCUSSION TITLE
'Line' object has no attribute 'get_field' and System' object has no attribute 'Breaker'

### DISCUSSION THREAD

**Discussion by TheASingh** [CURENT/andes#341] [General]
Hi All,

I was searching for a way to **break/disable** a `Line` dynamically while the TDS is running. I found that in Streaming.py, there is a function `handle_event(self, Event)`. 

I tried to call it within the Streaming file. I got an error in following line of code (line number 345 in Streaming.py):

```
bus = self.system.Line.get_field(
                    'bus1', ['Line_' + str(int(idx - 1))])[0]
```
**AttributeError: 'Line' object has no attribute 'get_field'**

Then to bypass it, I somehow used a static value for bus variable. Then I got the error in following line of code (line number 355 in Streaming.py)

`self.system.Breaker.insert(**param)`

**AttributeError: 'System' object has no attribute 'Breaker'**

Please help me understand why I'm facing these errors? Or is this function still a part of testing/RND?

**Comment by cuihantao** [comment]
@TheASingh  and @rana253, are you from the same research group? The recent questions seem closely related.

It looks like `streaming.py` is not up to date. We don't yet support inserting a new device after the system has been set up.

**Comment by TheASingh** [comment]
Today, I updated ANDES to version 1.7.3. But still Line object doesn't have attribute 'get_field' and System has no attribute 'Breaker'.

Please guide why I'm not able to successfully invoke the `handle_event(self, Event)` function.

**Reply by cuihantao** [reply]
The functions in streaming.py do not work now. Please see a recent addition to the documentation in Getting Started - Input formats - Disturbances.
